picodom.d.ts

```diff
@@ -1,26 +1,41 @@
-export interface VNode<Data> {
+export as namespace picodom
+
+export interface VirtualNode<Data> {
   tag: string
-  data?: Data
+  data: Data
   children: VElement<{}>[]
 }
 
 export interface Component<Data> {
-  (data?: Data, ...children: VElement<{}>[]): VNode<Data>
+  (data: Data, children: VElement<{}>[]): VirtualNode<any>
 }
 
-export type VElement<Data> = VNode<Data> | string
+export type VElement<Data> = VirtualNode<Data> | string
+
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  children?: Array<VElement<{}> | number>
+): VirtualNode<Data>
 
 export function h<Data>(
   tag: Component<Data> | string,
   data?: Data,
-  ...children: VElement<{}>[]
-): VNode<Data>
+  ...children: Array<VElement<{}> | number>
+): VirtualNode<Data>
 
 export function patch(
   parent: Element,
   element: Element | null,
-  oldNode: VNode<{}> | null,
-  node: VNode<{}>
+  oldNode: VirtualNode<{}> | null,
+  node: VirtualNode<{}>
 ): Element
 
-export as namespace picodom
+declare global {
+  namespace JSX {
+    interface Element<Data=any> extends VirtualNode<Data> {}
```

Comment:
I gave the generic argument `Data` a default type of `any` to allow people to use it as `JSX.Element` like in line 3 in jsx-test.ts, which other libraries (preact, react) allow.